js/ui/appActivation.js

```diff
@@ -56,6 +58,10 @@ const AppActivationContext = new Lang.Class({
         }
 
         if (showSplash) {
+            if (Main.overview.visible) {
+                this._hideWindows();
```

Comment:
Should probably be done inside `this.showSplash()`, and only if the logic there actually decides to show the splash.
But not when `showSplash()` is called from `DesktopClient`, as that is only called when one application launches another (e.g. a file manager opening a file)